src/initializers/log4js/json-appender.ts

```diff
@@ -4,11 +4,32 @@ const jsonAppender = layout => {
   return logEvent => {
     let event =
       logEvent.level.levelStr === 'ERROR' ? createErrorLog(layout, logEvent) : createValidLog(layout, logEvent);
-    var json = JSON.stringify(event);
+    const json = safeStringify(event);
     console.log(json);
   };
 };
 
+const safeStringify = event => {
+  try {
+    return JSON.stringify(event);
+  } catch (error) {
+    let seen = new WeakSet();
+
+    const circularJson = JSON.stringify(event, (key, value) => {
```

Comment:
Did you try util.inspect?

Reply:
Yes.
The reason I preferred a custom implementation is that utils.inspect documentation says that is intended for debugging purposes and not depend on it programmatically.
http://nodejs.org/api/util.html#util_util_inspect_object_options

Reply:
What is log4js using internally? What would it do if you passed it a circular object?

Reply:
It uses https://www.npmjs.com/package/flatted which is a library to handle circular dependencies. This is the commit:
https://github.com/log4js-node/log4js-node/commit/29c92211f2b4f030186392f000501d107ae4fd3d